usr.bin/sockstat/sockstat.c

```diff
@@ -1107,235 +1093,420 @@ sctp_path_state(int state)
+	cw->fib			= 3;
+	cw->splice_address	= 14;
+	cw->inp_gencnt	= 2;
+	cw->encaps		= 6;
+	cw->path_state	= 10;
+	cw->conn_state	= 10;
+	cw->stack		= TCP_FUNCTION_NAME_LEN_MAX;
+	cw->cc			= TCP_CA_NAME_MAX;
+
+	int n, len;
+	struct file *xf;
+	struct sock *s;
+	struct passwd *pwd;
+
+	for (xf = files, n = 0; n < nfiles; ++n, ++xf) {
+		if (xf->xf_data == 0)
+			continue;
+		if (opt_j >= 0 && opt_j != getprocjid(xf->xf_pid))
+			continue;
+		s = RB_FIND(socks_t, &socks,
+			&(struct sock){ .socket = xf->xf_data});
+		if (s == NULL || (!check_ports(s)))
+			continue;
+		s->shown = 1;
+		if (opt_n || 
+			(pwd = cap_getpwuid(cappwd, xf->xf_uid)) == NULL)
+			len = snprintf(NULL, 0, "%lu", (u_long)xf->xf_uid);
+		else
+			len = snprintf(NULL, 0, "%s", pwd->pw_name);
+		if (len > cw->user)
+			cw->user = len;
+		len = snprintf(NULL, 0, "%lu", (u_long)xf->xf_pid);
+		if (len > cw->pid)
+			cw->pid = len;
+		len = snprintf(NULL, 0, "%d", xf->xf_fd);
+		if (len > cw->fd)
+			cw->fd = len;
+	
+		calculate_sock_column_widths(cw, s);
+	}
+	if (opt_j >= 0)
+		return;
+	SLIST_FOREACH(s, &nosocks, socket_list) {
+		if (!check_ports(s))
+			continue;
+		calculate_sock_column_widths(cw, s);
+	}
+	RB_FOREACH(s, socks_t, &socks) {
+		if (s->shown)
+			continue;
+		if (!check_ports(s))
+			continue;
+		calculate_sock_column_widths(cw, s);
+	}
+}
+
+static void
+display_sock(struct sock *s, struct col_widths *cw, char *buf, size_t bufsize)
+{
+	struct addr *laddr, *faddr;
+	bool first;
+	laddr = s->laddr;
+	faddr = s->faddr;
+	first = true;
+
+	snprintf(buf, bufsize, "%s%s%s%s",
+		s->protoname,
+		s->vflag & INP_IPV4 ? "4" : "",
+		s->vflag & INP_IPV6 ? "6" : "",
+		(laddr != NULL && faddr != NULL &&
+		s->family == AF_UNIX && laddr->address.ss_len == 0 &&
+		faddr->conn == 0) ? " (not connected)" : "");
+	printf(" %-*s", cw->proto, buf);
+	while (laddr != NULL || faddr != NULL) {
+		if (s->family == AF_UNIX) {
+			if ((laddr == NULL) || (faddr == NULL))
+				errx(1, "laddr = %p or faddr = %p is NULL",
+					(void *)laddr, (void *)faddr);
+			if (laddr->address.ss_len > 0)
+				formataddr(&laddr->address, buf, bufsize);
+			else
+				strlcpy(buf, "??", bufsize);
+			printf(" %-*s", cw->local_addr, buf);
+			format_unix_faddr(faddr, buf, bufsize);
+			printf(" %-*s", cw->foreign_addr, buf);
+		} else {
+			if (laddr != NULL) 
+				formataddr(&laddr->address, buf, bufsize);
+			else
+				strlcpy(buf, "??", bufsize);
+			printf(" %-*s", cw->local_addr, buf);
+			if (faddr != NULL)
+				formataddr(&faddr->address, buf, bufsize);
+			else
+				strlcpy(buf, "??", bufsize);
+			printf(" %-*s", cw->foreign_addr, buf);
+		}
+		if (opt_A)
+			printf(" 0x%16lx", s->pcb);
+		if (opt_f) 
+			printf(" %*d", cw->fib, s->fibnum);
+		if (opt_I) {
+			if (s->splice_socket != 0) {
+				struct sock *sp;
+				sp = RB_FIND(socks_t, &socks, &(struct sock)
+					{ .socket = s->splice_socket });
+				if (sp != NULL)
+					formataddr(&sp->laddr->address,
+								buf, bufsize);
+			}
+			else
```

Comment:
style: cuddle the else
```suggestion
			} else
```